### PR TITLE
fix(stoneintg-924): intermittent unittest failure missing application…

### PR DESCRIPTION
**Pull request by jencull**
… index

## Maintainers will complete the following section

- [x] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [x] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/konflux-ci/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

**Review by jsztuka — APPROVED** [review]
lgtm

File: internal/controller/snapshot/snapshot_controller_test.go [hunk 1]
```diff
@@ -198,6 +198,15 @@ var _ = Describe("SnapshotController", func() {
...
 		Expect(k8sClient.Update(ctx, hasSnapshot)).Should(Succeed())
 
+		// Use Eventually to ensure the update has been applied
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{
+				Namespace: hasSnapshot.Namespace,
+				Name:      hasSnapshot.Name,
+			}, hasSnapshot)
+			return err == nil && hasSnapshot.Spec.Application == "im-a-ghost"
+		}, time.Second*20).Should(BeTrue())
```
Comment (kasemAlem) [inline internal/controller/snapshot/snapshot_controller_test.go:208]: would it be better if we set the interval poll argument
```suggestion
		}, time.Second*20, time.Second*2).Should(BeTrue())
```

Comment (jencull) [reply internal/controller/snapshot/snapshot_controller_test.go:208]: I can do that if you wish. I tested it extensively in a local container with this setup at the point where the test was failing (with the old code) and its not hitting that time out at the low point of resources. If setting a poll interval is best practice then no problem to do that. LMK.

**Review by dirgim — APPROVED** [review]
LGTM

**Review by kasemAlem — APPROVED** [review]

**Comment by kasemAlem** [comment]
LGTM

**Review by dheerajodha — APPROVED** [review]

**Review by Josh-Everett — APPROVED** [review]

**Pull request merged**